### PR TITLE
HCX-39 Allow to pass only some fields into hid.Hydra#updateSubscription method

### PR DESCRIPTION
**Pull request by dolfinus**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ v1.4.2 [unreleased]
 -------------------
 ### Features
 -   [#25](https://github.com/latera/camunda-ext/pull/25) Add Planado APIv2 connector
+-   [#32](https://github.com/latera/camunda-ext/pull/32) Allow to pass only some fields into hid.Hydra#updateSubscription method
```
Comment (sugarfree1) [inline CHANGELOG.md]: pass only *specific* fields

File: src/org/camunda/latera/bss/connectors/hid/hydra/Subscription.groovy [hunk 2]
```diff
@@ -110,10 +112,48 @@ trait Subscription {
...
+        quant              : subscription.n_quant,
+        payDay             : subscription.n_pay_day,
+        beginDate          : subscription.d_begin,
+        endDate            : subscription.d_end,
+        chargeLogEndDate   : subscription.d_charge_log_end
+      ]
+    }
 
+    if (notEmpty(input.parSubscriptionId)) {
+      LinkedHashMap parentSubscription = getSubscription(input.parSubscriptionId)
```
Comment (sugarfree1) [inline src/org/camunda/latera/bss/connectors/hid/hydra/Subscription.groovy]: looks like it is not used

**Comment by sugarfree1** [comment]
resolve conflicts

File: src/org/camunda/latera/bss/connectors/hid/hydra/Subscription.groovy (hunk 2 above)
Comment (dolfinus) [reply src/org/camunda/latera/bss/connectors/hid/hydra/Subscription.groovy]: What is not used?

Comment (sugarfree1) [reply src/org/camunda/latera/bss/connectors/hid/hydra/Subscription.groovy]: usually you don't need variable ```parentSubscription``` in such situations, in ruby it's something like
```
existingSubscription = existingSubscription.merge(getSubscription(input.parSubscriptionId).slice(:n_customer_id, :n_account_id, :n_doc_id, :n_object_id)) 
```

don't you have groovy way for this?

Comment (dolfinus) [reply src/org/camunda/latera/bss/connectors/hid/hydra/Subscription.groovy]: There is [subMap|http://docs.groovy-lang.org/latest/html/groovy-jdk/java/util/Map.html#subMap(java.lang.Object[])] method in groovy which can be used for this.

But input arguments of updateSubscription have different names from output of SELECT - compare `subscriptionId` and `n_subscription_id`. And there is no way easier than currently one to convert them properly.

**Comment by dolfinus** [comment]
Any updates?

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,8 +12,10 @@ v1.4.2 [unreleased]
 -   [#25](https://github.com/latera/camunda-ext/pull/25) Add Planado APIv2 connector
 -   [#38](https://github.com/latera/camunda-ext/pull/38) Add method for job updating into PlanadoV2 class
 -   [#33](https://github.com/latera/camunda-ext/pull/33) Allow to pass goodValueId into hid.Hydra#getGoodAddParamsBy
+-   [#32](https://github.com/latera/camunda-ext/pull/32) Allow to pass only specific fields into hid.Hydra#updateSubscription method
 -   [#24](https://github.com/latera/camunda-ext/pull/24) Add methods for files to document attaching
 -   [#39](https://github.com/latera/camunda-ext/pull/39) Add custom fields in create and update methods in PlanadoV2 class
+-   [#32](https://github.com/latera/camunda-ext/pull/32) Allow to pass only specific fields into hid.Hydra#updateSubscription method
```
Comment (sugarfree1) [inline CHANGELOG.md]: don't do it twice

**Pull request merged**